File: 1757-Recyclable-and-Low-Fat-Products-EASY/soln.py
import pandas as pd

def find_products(products: pd.DataFrame) -> pd.DataFrame:
    mask = (products['low_fats'] == 'Y') & (products['recyclable'] == 'Y')
    return products.loc[mask, ['product_id']]
    
    # df = products[
    #     (products['low_fats'] == 'Y') &
    #     (products['recyclable'] == 'Y')
    # ]
    # return df[['product_id']]

File: 1757-Recyclable-and-Low-Fat-Products-EASY/soln.scala
import org.apache.spark.sql.DataFrame
import org.apache.spark.sql.functions.col

object Solution {
  def findProducts(products: DataFrame): DataFrame = {
    products
      .filter(col("low_fats") === "Y" && col("recyclable") === "Y")
      .select("product_id")
  }

}
